Bring back max-width on inside pages

whit537 opened the pull request:
Not sure why we turned this off. Without it, the content is way too wide on inside pages on wide screens, leading to subpar results such as:

![screen shot 2017-03-28 at 10 04 23 am](https://cloud.githubusercontent.com/assets/134455/24409338/f0b47636-139d-11e7-8974-c89049b8fcf4.png)

whit537 commented:
Here's with `max-width` for comparison:

![screen shot 2017-03-28 at 10 05 02 am](https://cloud.githubusercontent.com/assets/134455/24409392/10ddd3d0-139e-11e7-9ef2-73c62ce39da3.png)

jdorfman commented:
@whit537 looks a lot better. @XhmikosR got any objections?

whit537 commented:
I did test it at different viewport widths during initial development, and did a quick check again this morning with this PR. Feel free to verify further. :)

XhmikosR commented:
There must be a reason I removed it back then, but it's been a long time and I can't recall the exact reason. :/

Can't do many things due to a faulty driver; if it's not urgent I should be up and running in 1 day.

whit537 commented:
Leaving a day or two for you to merge if you like, @XhmikosR. :-)

jdorfman commented:
@whit537 I will check this asap

jdorfman commented:
@whit537 after merging `master` into this branch I checked the new-ish sponsors page and it shrinks them down a lot. Not sure about the others but I know 2 that won't be thrilled. Thoughts?

![image](https://cloud.githubusercontent.com/assets/398230/24483285/e9416a12-14ab-11e7-93ce-faa2279e81c4.png)

XhmikosR commented:
Nope this change is bad hence why I removed it back then.

![screen shot 2017-03-30 at 09 07 35-fullpage](https://cloud.githubusercontent.com/assets/349621/24489918/b54a9d3c-1528-11e7-94b2-de80b88e0572.png)

whit537 commented:
Bad in Firefox, fine in Chrome/Safari. Will take a look ...

whit537 commented:
Rebased on latest master, was 5523995e607b3f90260033a1139d4ea810565311.

Firefox addressed in d8187c7d6ae1435ece0d575b09557fa0e5689a7b.

> I checked the new-ish sponsors page and it shrinks them down a lot.

Interesting. I'm seeing this at super-wide (in FF and Chrome). Will look at next ...

![screen shot 2017-03-30 at 10 21 42 am](https://cloud.githubusercontent.com/assets/134455/24508864/ce4fd5ea-1532-11e7-8a52-92e85d6d640c.png)

![screen shot 2017-03-30 at 10 21 47 am](https://cloud.githubusercontent.com/assets/134455/24508863/ce4e6fb6-1532-11e7-9c3b-21ae5cb2c84d.png)

whit537 commented:
Sponsors cleaned up in 79385b2557de064c2250e3ec2f4dbb5067696e5d. How are we doing here, @XhmikosR @jdorfman? Ready to merge?

whit537 commented:
Tested in Chrome 56 and FF 52 at various widths.

![mbeso4dulm](https://cloud.githubusercontent.com/assets/134455/24512476/ad0d16f4-153c-11e7-8695-8f8771db421a.gif)

jdorfman commented:
Looks good but the sizes of the logos are not in proportion. We also need to get @piamancini's sign off on the 1 sponsor per row. Please don't merge yet. Thanks @whit537

whit537 commented:
Alright, I've pushed 79385b2557de064c2250e3ec2f4dbb5067696e5d off as too complex, rebased on latest master (looks like there has been some movement on `sponsors.html` in the meantime, ya?) and added 9d075ce874e5a3849b38cbdb39ad60b8eb87de56 as low-hanging fruit to keep this PR in scope.

The sponsors page isn't perfect (below) but afaict it's the same in production. ¯\\\_(ツ)\_/¯

We good here? :-)

![c7trysov4y](https://cloud.githubusercontent.com/assets/134455/24556585/e2e3f452-1602-11e7-811d-599d51e9e9f4.gif)

XhmikosR commented:
I'll have a close look tomorrow morning GMT+2 time and let you know.

XhmikosR commented on _sass/_style.scss:
```diff
@@ -326,6 +325,7 @@ a {
     margin: 0 auto 102.4px;
   }
   .inside-page-content {
+    max-width: 538px; /* 576 - 51.2 - 51.2 */
```
The math here doesn't seem right. Did you mean to write something else in the comment?

whit537 replied:
You're right. Removed in c8125d7fae4e89c658d4a2e09e0f453367a633f7.

XhmikosR commented:
I looked at the branch. I still personally don't like the max-width approach. We have a `container` and `col-*` where we handle this stuff.

That being said, it's not my project so not my call to make. Just wanted to be noted that I wouldn't go with this personally.

On a side note, might be worth adding some styles for when the user hovers on the logos or the nav items too. Currently there's no visual indication.

whit537 commented:
@XhmikosR My goal is to bring back the width constraint on the body copy, in a way that works cross-browser. Do you have an alternate proposal for achieving that? I am not seeing how `container` and `col-*` handle the width constraint. Am I missing something?

XhmikosR commented:
Well they have a max width. That is their purpose.

Again, what you do is redundant IMO and looks bad in high resolutions. 500px or so is too small. We live in an era with 4K displays.

That being said, it's not my call to make. So, disregard my comments if you still want to merge this.

XhmikosR commented:
Also, http://getbootstrap.com/css/#grid-options

whit537 commented:
> We live in an era with 4K displays.

Just because we _can_ have really wide body copy doesn't mean we _should_. ;-) The reason to limit the width of the body copy is [legibility](https://baymard.com/blog/line-length-readability):

> Having the right amount of characters on each line is key to the **readability of your text**. It shouldn’t merely be your design that dictates the width of your text, it should also be a matter of legibility.
>
>The optimal line length for your body text is considered to be **50-60 characters** per line, including spaces (“Typographie”, E. Ruder). Other [sources](http://webtypography.net/2.1.2) suggest that up to 75 characters is acceptable.

> Also, http://getbootstrap.com/css/#grid-options

I haven't used Bootstrap before, and I'd rather not climb that learning curve on this PR. If you'd like to make an alternate PR that achieves the same result using Bootstrap, I say go for it! :-)

XhmikosR commented:
You **are** already using Bootstrap. Perhaps you don't know it but that is where container comes from.

Also, regarding legibility, sorry but this is just wrong. When I have to zoom in it to read something, you are not improving legibility.

Again, you can just ignore my comments. I simply chimed in because I was asked by @jdorfman to have a look at the website some time ago; I have no gain from all this so you can do whatever you prefer :)

whit537 commented:
> When I have to zoom in it to read something

That sounds like a font size rather than line length issue. No?

> you can do whatever you prefer

Yeah, ultimately this is up to @jdorfman do decide what he wants here. :-)

whit537 commented:
P.S. I looked back through early PRs (#1, #3) to see what the original design called for in terms of the inside pages. Nothing jumped out at me, I think we may have communicated about that in Slack. @jdorfman Maybe dig those up or ping the designer to give us some guidance here?

XhmikosR commented:
Yeah, my main issue is the custom grid. I don't like reinventing the wheel so hence why I added the (basic) Bootstrap components back then to get the site live.

Ultimately, the whole code should use Bootstrap's grid or any other framework's grid but it's something I don't have time to do. :/

Anyway, it's not a big deal. Just keep in mind screens are getting bigger and there is no reason to limit the max-width when there's so much space left.

jdorfman commented:
I'll review this tomorrow. Crazy day.

jdorfman commented:
Here is the bottom line. I agree with @whit537 on sticking with the original design. With that said the `Sponsor` page needs to have an exception. If it is too much work to make an exception then screw it, we kill `max-width` site wide even though it looks better.

whit537 commented:
As this PR stands, we [do](https://github.com/sustainers/website/pull/53#issuecomment-290740472) have an exception for the sponsors page. 👍

jdorfman commented:
@whit537 sorry my bad, merge this tomorrow AM just in case we need to revert back for some reason. Cool?

jdorfman commented:
Great job @whit537 and @XhmikosR 

Leave this open in case we need to revert for some reason.

whit537 commented:
Are we till hosting on GitHub pages? I'm not seeing this in production, and I'm seeing a warning on our GitHub Pages settings: "Domain does not resolve to the GitHub Pages server."

![screen shot 2017-04-06 at 1 33 09 pm](https://cloud.githubusercontent.com/assets/134455/24767651/ae80ee70-1acd-11e7-9f47-99b4e3cfbbe8.png)

XhmikosR commented:
If Couldflare is properly configured this should still work without any issues. Also, it might be something temporary on GitHub's side.

whit537 commented:
Not a deployment issue, as #75 was deployed fine:

![screen shot 2017-04-06 at 2 56 31 pm](https://cloud.githubusercontent.com/assets/134455/24770823/4b40fccc-1ad9-11e7-9990-6f7ffb504c53.png)

I give up. 😞
